src/action.cpp

```diff
@@ -1052,9 +1052,14 @@ action_id handle_main_menu()
     const input_context ctxt = get_default_mode_input_context();
     std::vector<uilist_entry> entries;
 
-    const auto REGISTER_ACTION = [&]( action_id name ) {
-        entries.emplace_back( name, true, hotkey_for_action( name, /*maximum_modifier_count=*/1 ),
-                              ctxt.get_action_name( action_ident( name ) ) );
+    const auto REGISTER_ACTION = [&]( action_id name, std::optional<char> kb = std::nullopt ) {
+        if( kb.has_value() ) {
+            entries.emplace_back( name, true, kb.value(),
+                                  ctxt.get_action_name( action_ident( name ) ) );
+        } else {
+            entries.emplace_back( name, true, hotkey_for_action( name, /*maximum_modifier_count=*/1 ),
```

Comment:
Maybe only set the hotkey to `kb.value()` when `hotkey_for_action` returns nothing? Because people might want to use their custom hotkey here.

Reply:
Can users set it? How? This is the Keybinding menu that opens when I press `?` while in "MAIN MENU"

1. main menu
   - ![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/13402666/1e45cc16-dfbc-468d-9a83-eaefac4c1db5)
2. `?`
3. keybinding screen - there is no `debug` option
   - ![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/13402666/47a701e7-c16c-4d39-868b-7413be2982d0)

Reply:
This differs from the "Debug menu" you can open from anywhere, still with no keybinding:
1. Just be, no menu opened
2. `?` to open the keybindings menu
3. 
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/13402666/617bec3b-463e-4473-854a-6b73a60558d9)

Reply:
The keybindings menu opened when you press `?` in the main menu is for `uilist` which uses the `UILIST` input context. When the code here registers the hotkey, it calls `hotkey_for_action` which uses the default mode input context, which is the same context used for opening the debug menu from the main game UI. So yes, setting the keybindings for `Debug menu` would change the hotkey in the main menu, unless my memory AND code analysis utterly fail.

Reply:
Fixed